js/ParticleModifiers.js

```diff
@@ -103,3 +107,26 @@ PHOTONS.EvenIntervalIndexModifier.prototype.update = function( particle, target
 
 }
 
+//=======================================
+// LoopingTimeIntervalIndex Modifier
+//=======================================
+
+PHOTONS.LoopingTimeIntervalIndexModifier = function( totalSteps, imagesPerSecond) {
+		PHOTONS.Modifier.call( this );
+		this.totalSteps = Math.floor( totalSteps || 1 );
+		this.timePerImage = 1 / imagesPerSecond;
+	}
+
+PHOTONS.LoopingTimeIntervalIndexModifier.prototype = Object.create( PHOTONS.Modifier.prototype );
+
+PHOTONS.LoopingTimeIntervalIndexModifier.prototype.update = function( particle, target ) {
+
+	// keep track of the atlas we started on so we can start randomly and progress normally afterwards
+	if ( particle.atlasStartIndex == undefined ) 
+		particle.atlasStartIndex = particle.atlasIndex.x;	

```

Comment:
Well first each particle needs a unique ID (I should have already implemented this TBH). It would be easy to add as part of the constructor:

```javascript

PHOTONS.Particle = function() {

	var _id = 0;

	return function() {
		this.id = _id++;
		this.age = 0;
		this.alive = 0;
		this.lifeSpan = 0;

		this.size = new THREE.Vector3();
		this.color = new THREE.Color();
		this.alpha = new PHOTONS.SingularVector( 0 );
		this.atlasIndex = new PHOTONS.SingularVector( 0 );

		this.position = new THREE.Vector3();
		this.velocity = new THREE.Vector3();
		this.acceleration = new THREE.Vector3();

		this.rotation = new PHOTONS.SingularVector( 0 );
		this.rotationalSpeed = new PHOTONS.SingularVector( 0 );
		this.rotationalAcceleration = new PHOTONS.SingularVector( 0 );

		this._tempVector3 = new THREE.Vector3();
	}

}();
```

Then, in `LoopingTimeIntervalIndexModifier` you'd have to implement a local data store with a custom data object linked to each particle. Maybe something like:

```javascript
PHOTONS.LoopingTimeIntervalIndexModifier = function( totalSteps, imagesPerSecond) {
		PHOTONS.Modifier.call( this );
		this.totalSteps = Math.floor( totalSteps || 1 );
		this.timePerImage = 1 / imagesPerSecond;
		this.customParticleData = {};
	}

PHOTONS.LoopingTimeIntervalIndexModifier.prototype = Object.create( PHOTONS.Modifier.prototype );

PHOTONS.LoopingTimeIntervalIndexModifier.prototype.update = function( particle, target ) {

	// keep track of the atlas we started on so we can start randomly and progress normally afterwards
	var customData = this.customParticleData[particle.id];
	if(!customData) {
		customData = {"atlasStartIndex": particle.atlasIndex.x};
		this.customParticleData[particle.id] = customData;
	}
	
	var step = Math.floor( customData.atlasStartIndex + particle.age / this.timePerImage ) % this.totalSteps;	
	target.set( step, step, step );

}
```